src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferDictionaryContainsMethods.cs

```diff
@@ -0,0 +1,223 @@
+                return;
+            if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCollectionsGenericIEnumerable1, out var ienumerableType))
+                return;
+
+            compilationContext.RegisterOperationAction(OnOperationAction, OperationKind.Invocation);
+
+            void OnOperationAction(OperationAnalysisContext context)
+            {
+                //  We report a diagnostic if we find an invocation of an applicable Contains method,
+                //  and the contains method is being invoked on an applicable property.
+
+                //  A property is applicable if:
+                //      1) It belongs to a type that implements IDictionary`2
+                //      2) It's name is either "Keys" or "Values"
+                //  A Contains method is applicable if:
+                //      1) It has a boolean return type.
+                //      2) It has one argument, not counting the first argument of an extension method.
+                //      3) The argument must match the applicable type argument.
+                //          - If the property's name is "Keys", it must match the type substituted for TKey in the IDictionary`2 instance.
+                //          - If the property's name is "Values", it must match the type substituted for TValue in the IDictionary`2 instance.
+
+                //  Once we have an applicable Contains method and an applicable property reference, we search for an applicable ContainsXXX
+                //  method on the IDictionary`2 receiver. 
+
+                //  A ContainsXXX method is applicable if:
+                //      1) It has a boolean return type
+                //      2) It is publically accessible
+                //      3) Its name is "ContainsKey" if property is "Keys", or "ContainsValue" if property is "Values". 
+                //      4) It has exactly one parameter of the correct type:
+                //          - If the method is a "ContainsKey" method, its type must be TKey. 
+                //          - If the method is a "ContainsValue" method, its type must be TValue.
+
+                var invocation = (IInvocationOperation)context.Operation;
+                IMethodSymbol containsMethod = invocation.TargetMethod;
+
+                if (containsMethod.Name != ContainsMethodName)
+                    return;
+                if (containsMethod.ReturnType.SpecialType != SpecialType.System_Boolean)
+                    return;
+                if (!TryGetPropertyReferenceOperation(invocation, out IPropertySymbol? property))
+                    return;
+                if (!DoesContainsCandidateHaveCorrectArgumentCount(containsMethod))
+                    return;
+                if (!TryGetConstructedDictionaryType(property.ContainingType, out INamedTypeSymbol? constructedDictionaryType))
```

Comment:
Suggestion:

```suggestion
                if (containsMethod.Name != ContainsMethodName
                    || containsMethod.ReturnType.SpecialType != SpecialType.System_Boolean
                    || !TryGetPropertyReferenceOperation(invocation, out IPropertySymbol? property)
                    || !DoesContainsCandidateHaveCorrectArgumentCount(containsMethod)
                    || !TryGetConstructedDictionaryType(property.ContainingType, out INamedTypeSymbol? constructedDictionaryType))
```

or with the `||` at the end of the line.

Reply:
You're right, that looks nicer. I'll change it.